term_proposal/t0037799/proposal.md

```diff
@@ -19,18 +19,30 @@ Quantized models from pytorch would be export to onnx format.
 
 So, I only need to implement 2 things which is the frontend and backend of my inference engine.
 
+Also, I'll provide a python interface for ease of use.
+
 ### 1. The interpreter for onnx operators
 onnx define a lot of operators for neural network model such as conv and maxpool.
 I need to parse the data, according to the parameters to build the corresponding model and pass it to my backend system.
 ### 2. The computing runtime.
 In the computing stage, the runtime need to manage memory, data flow and multi-threading.
 It should maximize the use of SIMD to speed up, and minimize the memory footprint.
 
----
+## Implementation
+
+The prototype will at least support mnist dataset on AlexNet.
+
+To support more operator and dataset, I'll define a generic abstract layer for them.
+User could define their own operator and dataset through inheritance
+
+Below is the architecture of the project.
 
-The prototype of the engine should be able to build a simple AlexNet model, so I would at least implement the basic operators need by AlexNet.
+![architecture](./architecture.svg)
 
-Furthermore, the dataloader and more operators(e.g. LSTM) could be support if I have free time.
+## Testing
+To ensure the correctness of this engine, I'll mainly test on 2 things.
+1. Test each operator's correctness with unittest.

```

Comment:
The number of operators significantly affect the efforts of the project.  I'd suggest you to list all of them and assess.